test/onnx/test_utility_funs.py

```diff
@@ -1044,6 +1044,137 @@ def forward(self, input, other):
+                seq_length = input_shape[1]
+
+                if position_ids is None:
+                    position_ids = torch.arange(
+                        seq_length, dtype=torch.long, device=device
+                    )
+                    position_ids = position_ids.unsqueeze(0).expand(input_shape)
+                if token_type_ids is None:
+                    token_type_ids = torch.zeros(
+                        input_shape, dtype=torch.long, device=device
+                    )
+                if inputs_embeds is None:
+                    inputs_embeds = self.word_embeddings(input_ids)
+
+                position_embeddings = self.position_embeddings(position_ids)
+                embeddings = inputs_embeds + position_embeddings
+
+                if self.token_type_embeddings:
+                    embeddings = embeddings + self.token_type_embeddings(token_type_ids)
+
+                embedding_output = CustomLayerNorm.apply(embeddings)
+                embedding_output = embedding_output * (
+                    attention_mask.unsqueeze(-1).type_as(embedding_output)
+                )
+                if attention_mask is None:
+                    attention_mask = torch.ones(input_shape, device=device)
+
+                query = embedding_output.transpose(0, 1)
+                tgt_len, bsz, embed_dim = query.size()
+                q, _, _ = self.in_proj(query).chunk(3, dim=-1)
+
+                head_dim = embed_dim // self.num_heads
+                scaling = (head_dim * 1) ** -0.5
+                q = (
+                    q.contiguous()
+                    .view(tgt_len, bsz * self.num_heads, head_dim)
+                    .transpose(0, 1)
+                    * scaling
+                )
+                return q
+
+        model = EmbeddingModule(
+            hidden_size, vocab_size, max_position_embeddings, type_vocab_siz
+        ).eval()
+        input_ids = torch.ones(batch_size, max_position_embeddings).long()
+        # 17 masks is referenced from the original transformer model
+        attention_mask = torch.cat(
+            ((torch.ones(17)), torch.zeros(max_position_embeddings - 17))
+        )
+        attention_masks = attention_mask.repeat(batch_size, 1)
+        f = io.BytesIO()
+        torch.onnx.export(
+            model,
+            (input_ids, attention_masks),
+            f,
+            opset_version=self.opset_version,
+            input_names=["input_ids", "attention_masks"],
+            dynamic_axes={"input_ids": [0, 1], "attention_masks": [0, 1]},
+            custom_opsets={"com.microsoft": 1},
+        )
+        graph = onnx.load(io.BytesIO(f.getvalue()))
+
+        # If there is a onnx::Constant node with dim=3, it is making the shape static
+        for node in graph.graph.node:
+            if node.op_type == "Constant":
+                if node.attribute[0].t.dims:
+                    self.assertNotEqual(node.attribute[0].t.dims[0], 3)
```

Comment:
I would assign 3 to a meaningfully named variable to avoid magic numbers

Reply:
Will change this

Reply:
Done